pycrunch/shoji.py

```diff
@@ -247,3 +267,29 @@ def graph(self, newgraph):
         """Update the Order with the new graph."""
         self['graph'] = newgraph
         super(Order, self).put(data=self.json)
+
+
+class TaskProgressTimeoutError(Exception):
+    def __init__(self, entity, response):
+        super(TaskProgressTimeoutError, self).__init__(
+            'Task Progress did not complete before timeout. '
+            'Trap this exception and call exc.entity.wait_progress(exc.response) '
```

Comment:
Surely we can do a better job of documenting this feature better than this.

And if we're offering a solution to tell users how to wait for completion without bailing on a timeout, why not just let them say `timeout=None` in their request instead of throwing an exception and catching it?